cpp/src/io/utilities/hostdevice_vector.hpp

```diff
@@ -66,19 +61,11 @@ class hostdevice_vector {
   {
     CUDF_EXPECTS(initial_size <= max_size, "initial_size cannot be larger than max_size");
 
-    if (hostdevice_vector_uses_pageable_buffer()) {
-      h_data_owner = thrust::host_vector<T>();
-    } else {
-      h_data_owner = cudf::detail::pinned_host_vector<T>();
-    }
-
-    std::visit(
-      [&](auto&& v) {
-        v.reserve(max_size);
-        v.resize(initial_size);
-        host_data = v.data();
-      },
-      h_data_owner);
+    h_data_owner = std::make_unique<cudf::detail::rmm_host_vector<T>>(
```

Comment:
Do you never want to access the pinned host memory from the device? Because your `cudf::host_resource_ref` only defines the `host_accessible` property, you are saying that the resource is only required to allocate host-accessible memory. But the pinned resource allocates memory that is accessible from host and device. So you could avoid memcopies when using it for host_device vectors...

Reply:
At this point we don't directly access the pinned memory from the device, and don't plan to. We measured that explicit copies are faster in cuIO.
Having a resource that only requires host access allows the resource to fall back to pageable memory if it's unable to provide a pinned memory buffer. This is the current behavior of the resource that spark plugin uses so everything lines up well.

Reply:
Got it.